lang/jvm/src/test/scala/com/wavesplatform/lang/compiler/DecompilerTest.scala

```diff
@@ -0,0 +1,260 @@
+import com.wavesplatform.lang.v1.compiler.{Decompiler, ExpressionCompilerV1, Terms}
+import com.wavesplatform.lang.v1.compiler.Terms._
+import com.wavesplatform.lang.v1.evaluator.ctx.impl.PureContext
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatest.prop.PropertyChecks
+
+class DecompilerTest extends PropSpec with PropertyChecks with Matchers {
+
+  val opcodes = scala.collection.immutable.Map[Int,String](1-> "+", 100->"sigVerify")
+
+  property("simple if") {
+    val expr = IF(TRUE, CONST_LONG(1), CONST_STRING("XXX"))
+    Decompiler(expr, opcodes) shouldBe
+      """{ if (
+        |    true
+        |    )
+        |then
+        |    1
+        |else
+        |    "XXX"
+        |}""".stripMargin
+  }
+
+  property("if with complicated else branch") {
+    val expr = IF(TRUE, CONST_LONG(1),
+      IF(TRUE, CONST_LONG(1),
+        CONST_STRING("XXX")))
+    Decompiler(expr, opcodes) shouldBe
+      """{ if (
+        |    true
+        |    )
+        |then
+        |    1
+        |else
+        |    { if (
+        |        true
+        |        )
+        |    then
+        |        1
+        |    else
+        |        "XXX"
+        |    }
+        |}""".stripMargin
+  }
+
+  property("if with complicated then branch") {
+    val expr = IF(TRUE,
+      IF(TRUE, CONST_LONG(1),
+        CONST_STRING("XXX")),
+      CONST_LONG(1))
+    Decompiler(expr, opcodes) shouldBe
+      """{ if (
+        |    true
+        |    )
+        |then
+        |    { if (
+        |        true
+        |        )
+        |    then
+        |        1
+        |    else
+        |        "XXX"
+        |    }
+        |else
+        |    1
+        |}""".stripMargin
+  }
+
+  property("simple let") {
+    val expr = Terms.BLOCKV1(LET("a", CONST_LONG(1)), TRUE)
+    Decompiler(expr, opcodes) shouldBe "{ let a = 1; true }"
+  }
+
+  property("native function call without args") {
+    val expr = Terms.FUNCTION_CALL(
+      function = FunctionHeader.Native(100),
+      args = List(TRUE)
+    )
+    Decompiler(expr, opcodes) shouldBe "sigVerify(true)"
+  }
+
+
+  property("undefined native function call without args") {
+    val expr = Terms.FUNCTION_CALL(
+      function = FunctionHeader.Native(101),
+      args = List(TRUE)
+    )
+    Decompiler(expr, opcodes) shouldBe "<Native_101>(true)"
+  }
+
+
+  property("user function call with one args") {
+    val expr = Terms.FUNCTION_CALL(
+      function = FunctionHeader.User("foo"),
+      args = List(TRUE)
+    )
+    Decompiler(expr, opcodes) shouldBe "foo(true)"
+  }
+
+  property("user function call with empty args") {
+    val expr = Terms.FUNCTION_CALL(
+      function = FunctionHeader.User("foo"),
+      args = List.empty
+    )
+    Decompiler(expr, opcodes) shouldBe "foo()"
+  }
+
+  property("user function declaration") {
+    val expr = Terms.FUNC("foo", List("bar", "buz"), CONST_BOOLEAN(true))
+    Decompiler(expr, opcodes) shouldBe "{ func foo (bar,buz) = TRUE }"
```

Comment:
func body is EXPR, i think it should be  `func foo (bar,buz) = { TRUE }` instead, no?